x-pack/plugins/security_solution/server/lib/detection_engine/signals/wrap_sequences_factory.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SearchAfterAndBulkCreateParams, WrappedSignalHit, WrapSequences } from './types';
+import { buildSignalGroupFromSequence } from './build_bulk_body';
+
+export const wrapSequencesFactory = ({
+  ruleSO,
+  signalsIndex,
+}: {
+  ruleSO: SearchAfterAndBulkCreateParams['ruleSO'];
+  signalsIndex: string;
+}): WrapSequences => (sequences) => {
+  const wrappedDocs = sequences.reduce(
+    (acc: WrappedSignalHit[], sequence) =>
+      acc.concat(buildSignalGroupFromSequence(sequence, ruleSO, signalsIndex)),
```

Comment:
nit: can we do a splat and avoid the `concat` here? Something like this? (warning I did not test this, but this is usually the format I see in the code base).

```ts
[...acc, buildSignalGroupFromSequence(sequence, ruleSO, signalsIndex)]
```